my-app/src/utils/WeatherApi.js

```diff
@@ -0,0 +1,54 @@
+class Api {
+  constructor(apiKey) {
+    this._apiKey = apiKey;
+  }
+
+  _handleResponse = (res) => {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error ${res.status}`);
+    }
+  };
+
+  getCityWeather(cityName) {
+    return fetch(
+      `/data/2.5/weather?q=${cityName}&units=imperial&appid=${this._apiKey}`,
+      {
+        headers: {
```

Comment:
```js

        headers: {
          Authorization: `${this._apiKey}`,
          'Content-Type': 'application/json',
        },

```

no need to repeate this object in every request. You need to pass it to the constructor and use it like this


```js


headers: this._headers,

```